flask_discord_interactions/models/modal.py

```diff
@@ -0,0 +1,74 @@
+from dataclasses import dataclass
+from typing import List
+
+from flask_discord_interactions.models.utils import LoadableDataclass
+from flask_discord_interactions.models.component import Component, ComponentType
+from flask_discord_interactions.enums import ResponseType
+
+
+@dataclass
+class Modal(LoadableDataclass):
```

Comment:
This class will need to be added to the documentation. You've written nice docstrings (which is most of the work honestly) but the class name needs to be added to a file in the `docs` directory for Sphinx to pick up on it. Ideally I'd like to have a brief written overview of using modals, some short code examples, and then the autodoc.

You can see a good example in the autocomplete docs, file is [here](https://github.com/Breq16/flask-discord-interactions/blob/664e70908a597c344eac2792422896de71b71194/docs/autocomplete.rst) and output is [here](https://flask-discord-interactions.readthedocs.io/en/latest/autocomplete.html).

If you're familiar with Sphinx/reStructuredText for docs or willing to learn, I'd encourage you to give them a try; if not, I'm happy to write up the page for this.

Reply:
I started a bit of a draft, but yeah, I might leave it up to you.